src/components/AddToCalendar/AddToCalendar.tsx

```diff
@@ -0,0 +1,82 @@
+import { ComponentProps, forwardRef, ElementType, useState } from 'react';
+import { getCommonProps, px } from '../../utils';
+import classnames from 'classnames';
+import * as Popover from '@radix-ui/react-popover';
+import { Icon } from '../Icon';
+import { CalendarEvent } from './types';
+import { Link, LinkProps } from '../Link';
+import {
+  generateGoogleCalendarLink,
+  generateOutlookOnlineLink,
+  generateYahooCalendarLink,
+  generateCalendarFile,
+} from './CalendarLinks';
+
+export interface AddToCalendarProps extends ComponentProps<'div'> {
+  event: CalendarEvent;
+  linkElement?: ElementType<LinkProps>;
+  label?: string;
+}
+
+/**
+ * ## Overview
+ *
+ * Overview of AddToCalendar component
+ *
+ * [Figma Link](https://www.figma.com/design/OvBXAq48blO1r4qYbeBPjW/RW---Sale-Page-(PLP)?node-id=4791-30655)
+ *
+ * [Storybook Link](https://deploy-preview-600--phillips-seldon.netlify.app/?path=/docs/components-addtocalendar--overview)
+ */
+
+const AddToCalendar = forwardRef<HTMLDivElement, AddToCalendarProps>(
+  ({ className, event, label = 'Add to calendar', linkElement: Component = Link, ...props }, ref) => {
+    const { className: baseClassName, ...commonProps } = getCommonProps(props, 'AddToCalendar');
+    const [open, setOpen] = useState(false); // State to manage the open/close of the popover
+
+    return (
+      <div {...commonProps} className={classnames(baseClassName, className)} {...props} ref={ref}>
+        <Popover.Root open={open} onOpenChange={setOpen}>
+          <Popover.Trigger asChild>
+            <button aria-label={label} className={`${px}-icon-button ${px}-icon-button--small`} type="button">
+              <Icon icon={open ? 'Close' : 'CalendarAlt'} />
+            </button>
+          </Popover.Trigger>
+          <Popover.Portal>
+            <Popover.Content sideOffset={5} align="start" side="bottom">
+              <ul className="atcb-list" role="dialog">
+                <li className="atcb-item">
+                  <button className="atcb-item-link" onClick={() => generateCalendarFile(event)}>
+                    iCalendar
```

Comment:
these need to be translatable and the text size needs to scale responsively.  We should be able to wrap this in a `<Text element="span"` component, looks like the Text variant in the Figma is `body2`